openhantek/src/hantekdso/modelspecification.h

```diff
@@ -31,19 +31,19 @@ struct ControlSamplerateLimits {
     std::vector<RecordLength> recordLengths; ///< Available record lengths
 
     inline double minSamplerate(RecordLengthID id) const {
-        return base / maxDownsampler / recordLengths[id].recordLength;
+        return base / maxDownsampler / recordLengths[id].bufferDivider;
```

Comment:
Record length is not the same as bufferDivider. For non-roll mode bufferDivider is 1, which makes sense: sample rate is a base frequency divided by the downsampler. recordLength should have nothing to do with it.